### PR TITLE
Improve save_path in AutoMM

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

1. For object detection, if model_loaded=True, every time calling `.predict()`  would create a new save folder and change `_save_path`.
2. Creating saving folder in `init()` complicates the logic of save_path.
3. Predictor and matcher use different save_path logic.

*Description of changes:*

1. Remove the creation of saving folder from `init()`. Create saving folder only when it's used.
2. Remove the `_model_loaded` flag. For saving the prediction results of object detection, we can directly save to the path `_save_path`, whether the model is loaded or fitted.
3. Let matcher use the same save_path logic as predictor.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by FANGAreNotGnu — APPROVED** [review]
LGTM, thanks for the fix!

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/src/autogluon/multimodal/utils/save.py [hunk 1]
```diff
@@ -139,10 +138,10 @@ def setup_save_path(
     elif proposed_save_path is not None:  # TODO: distinguish DDP and existed predictor
         save_path = process_save_path(path=proposed_save_path, raise_if_exist=(raise_if_exist and rank == 0))
     elif old_save_path is not None:
-        if model_loaded or fit_called:
+        if fit_called:
             save_path = process_save_path(path=old_save_path, raise_if_exist=False)
         else:
-            save_path = os.path.abspath(os.path.expanduser(old_save_path))
+            save_path = process_save_path(path=old_save_path, raise_if_exist=(raise_if_exist and rank == 0))
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/utils/save.py:144]: We may need a multi-GPU machine in the CI to test for multi-GPU training next time.

Comment (zhiqiangdon) [reply multimodal/src/autogluon/multimodal/utils/save.py:144]: Agree. Maybe select partial tests to avoid slow down CI.

**Review by bryanyzhu — APPROVED** [review]
Awesome, thanks for the fix.

**Pull request merged**